### ISSUE TITLE
More flexible profiling support

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7203]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#477|
|Type|enhancement|
|Created|2004-01-15 09:56:49Z|

Attachments:

* [newprof.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/866/8665243ef242a2b13ff662d0943d369e8ba0e206/4217ae74df59bea9e53edc66961cf1032d30802d.py) (1571 bytes) - added by exarkun on 2004-01-15 09:56:49Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__477 477
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__invalid invalid
component__core core
keywords__ 
time__1074160609000000 1074160609000000
changetime__1470699824468728 1470699824468728
version__ 
owner__ 
cc__glyph cc__radix cc__exarkun cc__itamarst cc__warner cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Currently, profiling can only be enabled when a process is started, and only
disabled by terminating that process.

Attached code is part of what would be required to let users enable, disable, or
switch between the two available profilers any time during the run of an
application.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Patch doesn't seem to work anymore.  Python's profilers don't like being started
and stopped like this.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub commented|
|-|-|

In changeset 5cfcea35a865bba6152b5dd51bc82a4cfb9e6898
```
#!CommitTicketReference repository="" revision="5cfcea35a865bba6152b5dd51bc82a4cfb9e6898"
Merge pull request #477 from twisted/8733-rodrigc-pollingfile-py3

Author: hawkowl, rodrigc
Reviewer: alex
Fixes: #8733

Fix twisted.internet.test.test_pollingfile on Python 3
```
